### DISCUSSION TITLE
Gradle War - Missing node_modules

### DISCUSSION THREAD

**Discussion by rafaelpsousa** [SCADA-LTS/Scada-LTS#3015] [General]
Due to a change in my code, I needed to recompile the Scada-LTS (gradle war) and in the generated war file was missing a Vue library folder, I had to add it manually (/resources/node_modules).

I miss something or is a gradle config issue?

**Comment by Limraj** [comment]
Hi @rafaelpsousa,
we recommend using gradle tasks(from Intellij), which contain all the necessary steps from compiling, running tests, to creating a war and deploying the application to tomcat, and starting tomcat itself:
https://github.com/SCADA-LTS/Scada-LTS/wiki/Getting-started-Development-%E2%80%90-Linux#51-gradle-tasks-
https://github.com/SCADA-LTS/Scada-LTS/wiki/Getting-started-Development-%E2%80%90-Windows#51-gradle-tasks-

**Reply by rafaelpsousa** [reply]
Hi @Limraj 

I followed your recommendation and generated the war through Intellij, the same problem occurred, in the generated file (Scada-LTS.war) the resources/node_modules folder is missing, it is necessary due to some imports such as:

```
/WebContent/WEB-INF/tags/page.tag
...
   <script src="resources/node_modules/stompjs/lib/stomp.min.js"></script>
    <script src="resources/node_modules/sockjs-client/dist/sockjs.min.js"></script>
    
/WebContent/WEB-INF/jsp/pointHierarchySLTS.jsp

<script src="resources/node_modules/bootstrap/dist/js/bootstrap.min.js"></script>
<script src="resources/node_modules/bootstrap3-dialog/dist/js/bootstrap-dialog.min.js"></script>
<script src="resources/node_modules/stompjs/lib/stomp.min.js"></script>
<script src="resources/node_modules/sockjs-client/dist/sockjs.min.js"></script>

**Reply by Limraj** [reply]
Hi @rafaelpsousa,
It depends on what build task you are using, and what branch are you trying to build the application from?

If you are on the release/2.8.0 branch, you still need to update this branch with the git pull command. In the buildRunDebug task, the installWebDependency subtask was missing:

```
task buildRunDebug(type: GradleBuild) {
tasks = ['clearTomcat', 'clearProject', 'installWebDependency', 'test', 'war', 'deployTomcat', 'runDebug']
}
```

To solve this, execute this commend:
`git pull`

It is important to remember this command when we checkout a branch, in order to update it with this command every time we checkout a branch.

**Comment by rafaelpsousa** [comment]
Hi @Limraj ,

The 'installWebDependency' subtask resolved the issue

Thanks a lot

**Reply by Limraj** [reply]
No, you just have to do something like this:
`git checkout release/2.8.0`
And:
`git pull`
That's all.
